R/oi_bicycle_parking.R

```diff
@@ -0,0 +1,37 @@
+#' Identify for bicycle parking.
+#'
+#' This function re-categorises OSM data, adding a column to an OSM network 
+#' `oi_cycle_parking` with value `"yes"` if a node/way within the OSM network has
+#' facilities for bicycle parking. 
+#'
+#' @param osm_sf - A `sf` and `data.frame` object containing OpenStreetMap 
+#' infrastructure data, obtained from the 
+#' [`osmextract`](https://github.com/ropensci/osmextract) function.
+#' @param remove - if `TRUE`, removes all nodes that are not suitable for 
+#' bicycle parking.
+#' @return - The provided `osm_sf` is returned with an additional column, 
+#' `oi_cycle_parking` containing the value `"yes"` if this feature allows 
+#' bicycle parking.
+#' @details - Note: the input `osm_sf` must containt the following tags:
+#' `c("amenity")`. Note that as most (if not all) amenity tags for bicycle 
+#' parking are assigned to nodes, the `osm_sf` must also contain nodes. (the
+#' layer `"points"` must be requested when using `osmextract::oe_get()`)
+#' @export oi_bicycle_parking 
+#' @examples
+#' internal_data = example_data
```

Comment:
Are these OSM ways?

Most parking spaces are points I think. Would like to see some stats on bicycle parking in different layers.

Reply:
For now, the eaxmple data is ways only. I added the amentity column with mutate so the checks (for now) would pass - I am going to update the exmaple data to contain nodes too. 

>Most parking spaces are points I think. Would like to see some stats on bicycle parking in different layers.

Indeed this is the case -  for a 2.5km circular buffer around leeds city centre there are 291 amenity=="bicycle_parking" tags, all belonging to Nodes.

I think worldwide - there are ~ 950 ways that contain amenity="bicycle_parking" (add taginfo link here). So, my current plan is to combine the "lines" & "points" layers into a single combined network for each LAD, and then create the data packs from this, so we can obtain bicycle parking features (and any other useful nodes)